src/angularComponent/ngcOmniboxController.js

```diff
@@ -30,4 +72,81 @@ export default class NgcOmniboxController {
 
     return true;
   }
+
+  /**
+   * Tries to update the index of the currently highlighted item to be the next item in the list. If
+   * we've reached the end, it'll loop back around and highlight the first item.
+   *
+   * @returns {Boolean} -- Index of the newly highlighted item
+   */
+  highlightPrevious() {
+    if (this.highlightedIndex > 0) {
+      this.highlightedIndex--;
+    } else {
+      this.highlightedIndex = this._suggestionElements.length - 1;
+    }
+
+    return this.highlightedIndex;
+  }
+
+  /**
+   * Tries to update the index of the currently highlighted item to be the previous item in the
+   * list. If we've reached the starrt, it'll loop back and highlight the last item.
+   *
+   * @returns {Boolean} -- Index of the newly highlighted item
+   */
+  highlightNext() {
+    if (this.highlightedIndex < this._suggestionElements.length - 1) {
+      this.highlightedIndex++;
+    } else {
+      this.highlightedIndex = 0;
+    }
+
+    return this.highlightedIndex;
+  }
+
+  /**
+   * Returns the overall index of suggestion HTML element in our list of suggestions.
+   *
+   * @param {HTMLElement} elem
+   * @returns {Number|Null}
+   */
+  getSuggestionItemIndex(elem) {
+    const index = this._cachedNodeList.indexOf(elem);
+    return index >= 0 ? index : null; // -1 is our index for nothing highlighted
+  }
+
+  _handleKeyDown(keyCode) {
+    if (keyCode === KEY.UP) {
+      this.highlightPrevious();
+    } else if (keyCode === KEY.DOWN) {
+      this.highlightNext();
+    } else if (keyCode === KEY.ESC) {
+      this.highlightedIndex = -1;
+    }
+
+    this._updateHighlightedItem();
+  }
+
+  _updateHighlightedItem() {
+    let selectedEl = this.$element[0].querySelector('[aria-selected]');
```

Comment:
This method *really* depends on the rest of the code not screwing up and selecting 2 elements at once. Maybe it's a good idea to be more defensive and use `.querySelectorAll()`?

P.S. It might be more readable to use `const` so there are two variables: `prevSelectedEl`, `newSelectedEl`.